File: frontend/src/types/expense.ts
/** 报销类型 */
export const ExpenseCategoryMap: Record<string, string> = {
  travel: '差旅费',
  office: '办公费',
  entertainment: '招待费',
  transport: '交通费',
  other: '其他',
}
export type ExpenseCategory = 'travel' | 'office' | 'entertainment' | 'transport' | 'other'

/** 报销状态（前后端共用） */
export const ExpenseStatusMap: Record<string, string> = {
  draft: '草稿',
  pending: '待审批',
  approved: '已通过',
  rejected: '已驳回',
}
export type ExpenseStatus = 'draft' | 'pending' | 'approved' | 'rejected'

/** 报销记录（对齐后端 ExpenseResponse） */
export interface ExpenseItem {
  id: number
  expenseNo: string
  title: string
  category: ExpenseCategory
  deptName: string
  budgetId: number | null
  budgetName: string
  amount: number
  status: ExpenseStatus
  applicant: string
  applyDate: string
  remark: string
  attachmentUrl: string
  createdAt: string
}

/** 新增/编辑表单（对齐后端 ExpenseCreate / ExpenseUpdate） */
export interface ExpenseForm {
  title: string
  category: ExpenseCategory
  deptName: string
  budgetId: number | null
  amount: number | null
  applyDate: string
  remark: string
}

/** 筛选查询参数 */
export interface ExpenseQuery {
  keyword?: string
  category?: ExpenseCategory
  deptName?: string
  status?: ExpenseStatus
  startDate?: string
  endDate?: string
  page: number
  pageSize: number
}

/** 统计摘要（对齐后端 ExpenseSummary） */
export interface ExpenseSummary {
  totalAmount: number
  pendingCount: number
  approvedCount: number
  rejectedCount: number
}

File: backend/app/schemas/expense.py
"""费用报销 Schema（与前端 types/expense.ts 字段完全对齐）"""
from pydantic import BaseModel, Field


class ExpenseCreate(BaseModel):
    """新增报销（对齐前端 ExpenseForm）"""
    title: str = Field(..., description="报销标题")
    category: str = Field(..., description="报销类型: travel/office/entertainment/transport/other")
    dept_name: str = Field(..., description="所属部门")
    budget_id: int | None = Field(None, description="关联预算ID（可选）")
    amount: float = Field(..., gt=0, description="报销金额(>0)")
    apply_date: str = Field(..., description="申请日期 YYYY-MM-DD")
    remark: str = Field("", description="备注说明")


class ExpenseUpdate(BaseModel):
    """编辑报销（对齐前端 ExpenseForm，所有字段可选）"""
    title: str | None = Field(None, description="报销标题")
    category: str | None = Field(None, description="报销类型")
    dept_name: str | None = Field(None, description="所属部门")
    budget_id: int | None = Field(None, description="关联预算ID")
    amount: float | None = Field(None, gt=0, description="报销金额")
    apply_date: str | None = Field(None, description="申请日期")
    remark: str | None = Field(None, description="备注说明")


class ExpenseResponse(BaseModel):
    """报销记录响应（对齐前端 ExpenseItem）"""
    id: int
    expense_no: str = ""
    title: str
    category: str
    dept_name: str
    budget_id: int | None = None
    budget_name: str = ""
    amount: float
    status: str = "pending"
    applicant: str = ""
    apply_date: str
    remark: str = ""
    attachment_url: str = ""
    created_at: str = ""


class ExpenseSummary(BaseModel):
    """报销统计摘要（对齐前端 ExpenseSummary）"""
    total_amount: float = 0.0
    pending_count: int = 0
    approved_count: int = 0
    rejected_count: int = 0